src/kernel/emulation/linux/misc/proc_info.c

```diff
@@ -645,20 +645,13 @@ static bool parse_smaps_firstline(
 
 static long _proc_pidinfo_pathinfo(int32_t pid, void* buffer, int32_t bufsize)
 {
-	char path[64];
 	struct vchroot_unexpand_args args;
 
-	__simple_sprintf(path, "/proc/%d/exe", pid);
-
-	memset(buffer, 0, bufsize);
-	int rv = sys_readlink(path, buffer, bufsize - 1);
+	int rv = dserver_rpc_get_executable_path(pid, args.path, sizeof(args.path));
 
 	if (rv < 0)
 		return rv;
```

Comment:
Like I said [here](https://github.com/darlinghq/darlingserver/pull/1/files#r842329980), negative error codes from RPC calls indicate internal errors; we should abort on internal errors (maybe handling this should be moved into the RPC wrapper code later). However, positive error codes are okay (e.g. in this case, it'd okay to get positive `ESRCH`).
`_proc_pidinfo_pathinfo` is expected to return negative error codes, however, so when we get a positive error code from the server, we can negate it and return it to the user:
```suggestion
	if (rv < 0) {
		__simple_printf("dserver_rpc_get_executable_path failed internally: %d\n", rv);
		__simple_abort();
	} else if (rv > 0) {
		return -rv;
	}
```